### PR TITLE
[codex] Support negative floating-rate differentials

### PR DESCRIPTION
**Pull request by ajmatsick**
## Summary
- Allow negative loan-level floating-rate differentials while preserving required and product min/max validation.
- Allow linked floating products to configure negative product differential ranges and validate min/default/max ordering without non-negative assumptions.
- Add focused provider validator tests and MNZL schedule tests proving negative spreads and forced floating DTO behavior keep future floating-rate periods applicable.

## Behavior
Effective floating rates remain additive:

`floating/base period rate + product differential + loan differential`

No API field names changed and no migration is included. MNZL products must configure `minDifferentialLendingRate` below zero, for example `-5.00`, for a loan-level discount such as `-0.25` to pass validation.

## Data Audit
Read-only staging data audit checked 3 linked floating-rate product configurations; none violate product differential range or min/default/max ordering validation.

## Verification
- `./gradlew :fineract-provider:test --tests '*LoanApplicationValidatorTest' --tests '*LoanProductDataValidatorTest' :custom:mnzl:loan:schedule-api:test --tests '*MnzlLoanUtilServiceTest' --tests '*MnzlLoanTermVariationsEnricherTest'`
- `./gradlew :fineract-provider:spotbugsTest`
- `./gradlew :fineract-provider:spotlessCheck :custom:mnzl:loan:schedule-api:spotlessCheck`

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java [hunk 1]
```diff
@@ -2247,6 +2209,28 @@ public void validateMinMaxConstraints(final JsonElement element, final DataValid
...
+        }
+        if (maxDifferentialLendingRate != null && defaultDifferentialLendingRate != null) {
+            baseDataValidator.reset().parameter(MAX_DIFFERENTIAL_LENDING_RATE).value(maxDifferentialLendingRate)
+                    .notLessThanMin(defaultDifferentialLendingRate);
+        }
+        if (interestRateDifferential != null && minDifferentialLendingRate != null && maxDifferentialLendingRate != null
+                && minDifferentialLendingRate.compareTo(maxDifferentialLendingRate) <= 0) {
+            baseDataValidator.reset().parameter(INTEREST_RATE_DIFFERENTIAL).value(interestRateDifferential)
+                    .inMinAndMaxAmountRange(minDifferentialLendingRate, maxDifferentialLendingRate);
+        }
```
[the remark this answers is not in the document]
Comment (ajmatsick) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java:2231]: Addressed with validateForUpdate coverage for partial max narrowing, and updated the PR description with the read-only staging data audit summary: 3 linked floating-rate product configurations checked, 0 invalid.

**Comment by ajmatsick** [comment]
@greptileai
